### DISCUSSION TITLE
Aggressive Cache (Offline Mode)

### DISCUSSION THREAD

**Discussion by maltfield** [aeharding/voyager#1147] [Ideas]
** Describe the feature request below **

Feature Request: Please add an option to Voyager to "Aggressively Cache" content (with secondary toggle on/off for caching images)

## Why?

I do a lot of traveling, and I often don't have Internet access. Every few days I may visit a grocery store or a cafe with WiFi, and just after I connect I'll kick-off some downloads on my phone for news, podcasts, email, etc.

Then these apps will download their content in the background (to a cache) while I shop. And after I leave the shop, I can re-visit these apps from my tent where I don't have internet access.

## Solution

Voyager should have an option to cache content in the background. The goal would be that, when enabled, I would just open the app and pull down from the top (or click "refresh") and then lock my phone's screen. But in the background, Voyager will be downloading the top 100 posts of every community that I'm subscribed-to, including the post's body and all of its comments.

Then, hours later when my phone doesn't have internet access, I should be able to browse through all of my communities and view the downloaded posts in Voyager.

I think the following settings would be important for this feature:

1. Cache content for offline use (y/n)?
2. Cache size (MB)
3. Cache images (y/n)?
4. Cache videos (y/n)?

For an example of how this is implemented in another app, see [RedReader](https://github.com/QuantumBadger/RedReader).

**Comment by maltfield** [comment]
See also this [similar feature request in jerboa](https://github.com/dessalines/jerboa/issues/621)

 * https://github.com/dessalines/jerboa/issues/621

**Comment by aeharding** [comment]
Hi there! This isn't something I'm planning on implementing in the forseeable future. However I'll convert to a discussion so it can be discussed and maybe reconsidered in the future.

**Comment by maltfield** [comment]
If you don't personally have the capacity to work on this feature, that's fine. But to just close the ticket for a user-contributed feature request is rude and doesn't seem appropriate.

I think it would be better to mark the original feature request ticket with labels `low-priority` and `PRs welcome` rather than just completely close it and lock it.

**Reply by aeharding** [reply]
This project leans towards closing tickets more aggressively than many other projects. Please don't take it personally. This ensures that issues do not languish and become stale. Github Discussions is recommended for these topics instead of Issues.

**Comment by lullis** [comment]
Given this is still open, I assume it's not taken as high priority. However, I'd like to work on making Voyager a completely offline-first application. Would that be something you'd be interested on?

**Comment by aeharding** [comment]
Hey, so I've been revisiting longer term features, and this is one of them! There's no timeline for starting work, but I created issue #2104, please feel free to subscribe there for updates.
